source/TextArea.cpp

```diff
@@ -147,27 +149,29 @@ void TextArea::Draw()
 
 	const Point UP{0, -1};
 	const Point DOWN{0, 1};
+	const float SCROLLBAR_OFFSET = 5;
 	const float POINTER_OFFSET = 5;
 	if(scroll.Scrollable())
 	{
-		// Draw up and down pointers, mostly to indicate when scrolling
-		// is possible, but might as well make them clickable too.
-		Rectangle topRight(position + Point{buffer->Right(), buffer->Top() + POINTER_OFFSET}, {20.0, 20.0});
-		PointerShader::Draw(topRight.Center(), UP,
-			10.f, 10.f, 5.f, Color(scroll.IsScrollAtMin() ? .2f : .8f, 0.f));
-		AddZone(topRight, [&]() { scroll.Scroll(-Preferences::ScrollSpeed()); });
-
-		Rectangle bottomRight(position + Point{buffer->Right(), buffer->Bottom() - POINTER_OFFSET}, {20.0, 20.0});
-		PointerShader::Draw(bottomRight.Center(), DOWN,
-			10.f, 10.f, 5.f, Color(scroll.IsScrollAtMax() ? .2f : .8f, 0.f));
-		AddZone(bottomRight, [&]() { scroll.Scroll(Preferences::ScrollSpeed()); });
+		Rectangle topRight(position + Point{buffer->Right() + SCROLLBAR_OFFSET, buffer->Top() + POINTER_OFFSET}, {20.,
+			20.});
+		Rectangle bottomRight(position + Point{buffer->Right() + SCROLLBAR_OFFSET, buffer->Bottom() - POINTER_OFFSET},
+			{20., 20.});
+
+		scrollBar.SyncDraw(scroll, topRight.Center(), bottomRight.Center());
```

Comment:
these rectangles aren't reused later, and you only need their centers, so just use points
```suggestion
		Point topRight(position + Point{buffer->Right() + SCROLLBAR_OFFSET, buffer->Top() + POINTER_OFFSET});
		Point bottomRight(position + Point{buffer->Right() + SCROLLBAR_OFFSET, buffer->Bottom() - POINTER_OFFSET});

		scrollBar.SyncDraw(scroll, topRight, bottomRight);
```